url7d636e8121_laforge49_JActor2_tgz-pJ8-org_agilewiki_jactor2_core_revisited_EHJ8/src/org/agilewiki/jactor2/core/impl/mtMessages/RequestMtImpl.java

```
@@ -0,0 +1,623 @@
+     */
+    public boolean _isCanceled() {
+        return canceled;
+    }
+
+    public boolean isComplete() {
+        return !incomplete;
+    }
+
+    public IsolationReactor getIsolationReactor() {
+        return isolationReactor;
+    }
+
+    @Override
+    public void close() {
+        if (!incomplete) {
+            return;
+        }
+        incomplete = false;
+        closed = true;
+        response = new ReactorClosedException();
+        if (requestSource != null) {
+            requestSource.incomingResponse(this, null);
+        }
+    }
+
+    /**
+     * Cancel this request.
+     */
+    public void cancel() {
+        if (canceled) {
+            return;
+        }
+        canceled = true;
+    }
+
+    /**
+     * Process a request or the response.
+     */
+    public void eval() {
+        if (incomplete) {
+            targetReactorImpl.setExceptionHandler(null);
+            targetReactorImpl.setCurrentRequest(this);
+            targetReactorImpl.requestBegin(this);
+            try {
+                if (isolationReactor != null) {
+                    IsolationReactorMtImpl isolationReactorMtImpl =
+                            (IsolationReactorMtImpl) isolationReactor.asReactorImpl();
+                    isolationReactorMtImpl.addResource(targetReactorImpl);
+                }
+                processRequestMessage();
+            } catch (final MigrationException _me) {
+                throw _me;
+            } catch (final InterruptedException ex) {
+                Thread.currentThread().interrupt();
+            } catch (final RuntimeException re) {
+                processException(targetReactorImpl,
+                        new ReactorClosedException(re));
+                targetReactorImpl.getRecovery().onRuntimeException(this, re);
+            } catch (final Exception e) {
+                processException(targetReactorImpl, e);
+            } catch (final StackOverflowError soe) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)